src/components/react/ChessStudy.tsx

```diff
@@ -107,56 +104,62 @@ export const ChessStudy = ({
 					return draft;
 				}
 				case 'REMOVE_LAST_MOVE_FROM_HISTORY': {
-					if (!chessView || !draft || draft.study.moves.length == 0) return draft;
+					if (!chessView || hasNoMoves) return draft;
 
-					let moves = draft.study.moves;
+					const moves = draft.study.moves;
 
 					const currentMoveId = draft.currentMove?.moveId;
 
-					const { variant, moveIndex } = findMoveIndex(moves, currentMoveId);
+					if (currentMoveId) {
```

Comment:
If there is no `currentMoveId`, we should probably just select the last move of the main line and remove that one.